app/src/main/java/com/github/se/studentconnect/ui/profile/OrganizationProfileViewModel.kt

```
@@ -303,4 +319,135 @@ class OrganizationProfileViewModel(
+                ?: mutableSetOf<String>().apply { organization?.createdBy?.let { add(it) } }
+
+        // Filter out existing members and current user
+        val availableUsers =
+            allUsers.filter { user ->
+              !memberUids.contains(user.userId) && user.userId != currentUserId
+            }
+
+        _uiState.value =
+            _uiState.value.copy(availableUsers = availableUsers, isLoadingUsers = false)
+      } catch (e: Exception) {
+        Log.e(TAG, "Failed to load available users", e)
+        _uiState.value = _uiState.value.copy(isLoadingUsers = false)
+      }
+    }
+  }
+
+  /**
+   * Sends a member invitation to a user.
+   *
+   * @param userId The ID of the user to invite
+   */
+  fun sendMemberInvitation(userId: String) {
+    val currentOrg = _uiState.value.organization ?: return
+    val role = _uiState.value.selectedRole ?: return
+    val inviterId = currentUserId ?: return
+
+    viewModelScope.launch {
+      try {
+        // Get inviter name
+        val inviter = userRepository.getUserById(inviterId)
+        val inviterName = inviter?.getFullName() ?: "Unknown"
+
+        // Send invitation
+        organizationRepository.sendMemberInvitation(
+            organizationId = currentOrg.organizationId,
+            userId = userId,
+            role = role,
+            invitedBy = inviterId)
+
+        // Create notification
+        val notification =
+            Notification.OrganizationMemberInvitation(
+                id = "",
+                userId = userId,
+                timestamp = Timestamp.now(),
+                isRead = false,
+                organizationId = currentOrg.organizationId,
+                organizationName = currentOrg.name,
+                role = role,
+                invitedBy = inviterId,
+                invitedByName = inviterName)
+
+        notificationRepository.createNotification(notification, onSuccess = {}, onFailure = {})
+
```

Notification failure is silently ignored